client/src/track.ts

```diff
@@ -259,9 +259,10 @@ export default class Track {
         Math.round(feature.bounds[3]),
       ];
     }
-    if (feature.keyframe) {
-      listInsert(this.featureIndex, feature.frame);
+    if (!this.features[feature.frame].keyframe) {
+      throw new Error('setFeature must be called with keyframe=true OR to update an existing keyframe');
     }
+    listInsert(this.featureIndex, feature.frame);
```

Comment:
@BryonLewis this is a really important change that could have consequences if I'm wrong about it.

I've tested it myself, and it looks like, by coincidence, this condition was already always true, but you could call setFeature without a keyframe and it wouldn't be marked as keyframe=true, or get added to the index, but it would behave like one.  Added explicit error and associated unit test.

Please check my work here.

Reply:
Basically, the json2csv serializer assumes (and indeed, the python version does) that all features found in the json are keyframes.